proposals/dao/fip_104b.ts

```diff
@@ -0,0 +1,67 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import {
+  DeployUpgradeFunc,
+  NamedAddresses,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '@custom-types/types';
+import { time } from '@test/helpers';
+import { BigNumber } from 'ethers';
+
+/*
+Withdraw liquidity from the LBP once it has finished
+*/
+
+const fipNumber = 'fip_104b';
+
+let initialDaiPCVBalance: BigNumber;
+let initialTCDpiBalance: BigNumber;
+
+// Do any deployments
+// This should exclusively include new contract deployments
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  console.log(`No deploy actions for fip${fipNumber}`);
+  return {
+    // put returned contract objects here
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  const dpi = contracts.dpi;
+  initialTCDpiBalance = await dpi.balanceOf(addresses.tribalCouncilSafe);
+  initialDaiPCVBalance = await contracts.compoundDaiPCVDeposit.balance();
+  logging && console.log('Initial dai balance: ', initialDaiPCVBalance.toString());
+
+  // Fast forward to end of LPB
+  const timeRemaining = await contracts.dpiToDaiLBPSwapper.remainingTime();
+  await time.increase(timeRemaining);
+};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for fip${fipNumber}`);
+};
+
+// Run any validations required on the fip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  // 1. Validate withdrawn liquidity destinations
+  const sanityCheckDAIAmount = ethers.constants.WeiPerEther.mul(3_500_000);
```

Comment:
that's ~3% away from the current pool state, if DPI price changes too much before your proposal executes, the e2e setup step will fail, I think you should be more laxist on your sanity checks. For DPI for instance you checked balance of at least 1,500 where the spot pool content is 2,000, this gives more margin. Not a big deal but I wanted to share this with you, we had to update e2es because of this before

Reply:
That makes sense. I'll be less strict in the future on sanity checks